{"relationshipsSections":[{"identifiers":["doc:\/\/co.pointfree.VaporRouting\/s5ErrorP","doc:\/\/co.pointfree.VaporRouting\/s8SendableP"],"title":"Conforms To","kind":"relationships","type":"conformsTo"}],"metadata":{"navigatorTitle":[{"text":"StreamError","kind":"identifier"}],"externalID":"s:8NIOHTTP20A6ErrorsO11StreamErrorV","roleHeading":"Structure","modules":[{"name":"NIOHTTP2"}],"symbolKind":"struct","fragments":[{"text":"struct","kind":"keyword"},{"text":" ","kind":"text"},{"text":"StreamError","kind":"identifier"}],"title":"NIOHTTP2Errors.StreamError","role":"symbol"},"primaryContentSections":[{"kind":"declarations","declarations":[{"languages":["swift"],"platforms":["Linux"],"tokens":[{"kind":"keyword","text":"struct"},{"kind":"text","text":" "},{"kind":"identifier","text":"StreamError"}]}]},{"kind":"content","content":[{"anchor":"overview","type":"heading","text":"Overview","level":2},{"type":"paragraph","inlineContent":[{"text":"Stream errors are wrappers around another error of some other kind that occurred on a specific stream.","type":"text"},{"text":" ","type":"text"},{"text":"As they are a wrapper error, they carry a “real” error in ","type":"text"},{"identifier":"doc:\/\/co.pointfree.VaporRouting\/documentation\/NIOHTTP2\/NIOHTTP2Errors\/StreamError\/baseError","type":"reference","isActive":true},{"text":". Additionally, they cannot","type":"text"},{"text":" ","type":"text"},{"text":"meaningfully be ","type":"text"},{"code":"Equatable","type":"codeVoice"},{"text":", so they aren’t. There’s also no additional location information: that’s","type":"text"},{"text":" ","type":"text"},{"text":"provided by the base error.","type":"text"}]}]}],"variants":[{"paths":["\/documentation\/niohttp2\/niohttp2errors\/streamerror"],"traits":[{"interfaceLanguage":"swift"}]}],"identifier":{"url":"doc:\/\/co.pointfree.VaporRouting\/documentation\/NIOHTTP2\/NIOHTTP2Errors\/StreamError","interfaceLanguage":"swift"},"sections":[],"schemaVersion":{"patch":0,"minor":3,"major":0},"hierarchy":{"paths":[["doc:\/\/co.pointfree.VaporRouting\/documentation\/NIOHTTP2","doc:\/\/co.pointfree.VaporRouting\/documentation\/NIOHTTP2\/NIOHTTP2Errors"]]},"topicSections":[{"identifiers":["doc:\/\/co.pointfree.VaporRouting\/documentation\/NIOHTTP2\/NIOHTTP2Errors\/StreamError\/baseError","doc:\/\/co.pointfree.VaporRouting\/documentation\/NIOHTTP2\/NIOHTTP2Errors\/StreamError\/description","doc:\/\/co.pointfree.VaporRouting\/documentation\/NIOHTTP2\/NIOHTTP2Errors\/StreamError\/streamID"],"title":"Instance Properties"}],"kind":"symbol","abstract":[{"text":"A StreamError was hit during outbound frame processing.","type":"text"}],"references":{"doc://co.pointfree.VaporRouting/s8SendableP":{"type":"unresolvable","identifier":"doc:\/\/co.pointfree.VaporRouting\/s8SendableP","title":"Swift.Sendable"},"doc://co.pointfree.VaporRouting/documentation/NIOHTTP2/NIOHTTP2Errors/StreamError/baseError":{"type":"topic","abstract":[{"text":"The underlying thrown error.","type":"text"}],"fragments":[{"text":"var","kind":"keyword"},{"text":" ","kind":"text"},{"text":"baseError","kind":"identifier"},{"text":": ","kind":"text"},{"preciseIdentifier":"s:s5ErrorP","text":"Error","kind":"typeIdentifier"}],"identifier":"doc:\/\/co.pointfree.VaporRouting\/documentation\/NIOHTTP2\/NIOHTTP2Errors\/StreamError\/baseError","role":"symbol","kind":"symbol","title":"baseError","url":"\/documentation\/niohttp2\/niohttp2errors\/streamerror\/baseerror"},"doc://co.pointfree.VaporRouting/documentation/NIOHTTP2/HTTP2StreamID":{"url":"\/documentation\/niohttp2\/http2streamid","navigatorTitle":[{"text":"HTTP2StreamID","kind":"identifier"}],"fragments":[{"kind":"keyword","text":"struct"},{"kind":"text","text":" "},{"text":"HTTP2StreamID","kind":"identifier"}],"kind":"symbol","type":"topic","abstract":[{"type":"text","text":"A single HTTP\/2 stream ID."}],"role":"symbol","identifier":"doc:\/\/co.pointfree.VaporRouting\/documentation\/NIOHTTP2\/HTTP2StreamID","title":"HTTP2StreamID"},"doc://co.pointfree.VaporRouting/documentation/NIOHTTP2/NIOHTTP2Errors/StreamError":{"role":"symbol","title":"NIOHTTP2Errors.StreamError","navigatorTitle":[{"text":"StreamError","kind":"identifier"}],"url":"\/documentation\/niohttp2\/niohttp2errors\/streamerror","type":"topic","identifier":"doc:\/\/co.pointfree.VaporRouting\/documentation\/NIOHTTP2\/NIOHTTP2Errors\/StreamError","kind":"symbol","fragments":[{"text":"struct","kind":"keyword"},{"text":" ","kind":"text"},{"text":"StreamError","kind":"identifier"}],"abstract":[{"text":"A StreamError was hit during outbound frame processing.","type":"text"}]},"doc://co.pointfree.VaporRouting/documentation/NIOHTTP2/NIOHTTP2Errors/StreamError/description":{"type":"topic","role":"symbol","kind":"symbol","title":"description","url":"\/documentation\/niohttp2\/niohttp2errors\/streamerror\/description","identifier":"doc:\/\/co.pointfree.VaporRouting\/documentation\/NIOHTTP2\/NIOHTTP2Errors\/StreamError\/description","abstract":[],"fragments":[{"text":"var","kind":"keyword"},{"text":" ","kind":"text"},{"text":"description","kind":"identifier"},{"text":": ","kind":"text"},{"preciseIdentifier":"s:SS","text":"String","kind":"typeIdentifier"}]},"doc://co.pointfree.VaporRouting/documentation/NIOHTTP2/NIOHTTP2Errors":{"url":"\/documentation\/niohttp2\/niohttp2errors","kind":"symbol","type":"topic","fragments":[{"text":"enum","kind":"keyword"},{"kind":"text","text":" "},{"kind":"identifier","text":"NIOHTTP2Errors"}],"navigatorTitle":[{"kind":"identifier","text":"NIOHTTP2Errors"}],"abstract":[{"type":"text","text":"Errors that "},{"isActive":true,"type":"reference","identifier":"doc:\/\/co.pointfree.VaporRouting\/documentation\/NIOHTTP2"},{"type":"text","text":" raises when handling HTTP\/2 connections."}],"role":"symbol","identifier":"doc:\/\/co.pointfree.VaporRouting\/documentation\/NIOHTTP2\/NIOHTTP2Errors","title":"NIOHTTP2Errors"},"doc://co.pointfree.VaporRouting/documentation/NIOHTTP2/NIOHTTP2Errors/StreamError/streamID":{"abstract":[{"type":"text","text":"The "},{"identifier":"doc:\/\/co.pointfree.VaporRouting\/documentation\/NIOHTTP2\/HTTP2StreamID","isActive":true,"type":"reference"},{"type":"text","text":" on which the error was thrown."}],"url":"\/documentation\/niohttp2\/niohttp2errors\/streamerror\/streamid","kind":"symbol","fragments":[{"text":"var","kind":"keyword"},{"text":" ","kind":"text"},{"text":"streamID","kind":"identifier"},{"text":": ","kind":"text"},{"text":"HTTP2StreamID","preciseIdentifier":"s:8NIOHTTP213HTTP2StreamIDV","kind":"typeIdentifier"}],"role":"symbol","title":"streamID","type":"topic","identifier":"doc:\/\/co.pointfree.VaporRouting\/documentation\/NIOHTTP2\/NIOHTTP2Errors\/StreamError\/streamID"},"doc://co.pointfree.VaporRouting/documentation/NIOHTTP2":{"url":"\/documentation\/niohttp2","kind":"symbol","type":"topic","abstract":[],"role":"collection","identifier":"doc:\/\/co.pointfree.VaporRouting\/documentation\/NIOHTTP2","title":"NIOHTTP2"},"doc://co.pointfree.VaporRouting/s5ErrorP":{"type":"unresolvable","identifier":"doc:\/\/co.pointfree.VaporRouting\/s5ErrorP","title":"Swift.Error"}}}